apps/storefront/src/types/quotes.ts

```diff
@@ -88,11 +88,54 @@ export interface QuoteItem {
   };
 }
 
+export interface QuoteExtraFields {
+  id?: number | string;
+  fieldName: string;
+  value: string | number;
+}
+
 export interface QuoteInfo {
   userId?: number;
   contactInfo: ContactInfo;
   shippingAddress: ShippingAddress;
   billingAddress: BillingAddress;
   fileInfo?: FileInfo[];
   note?: string;
+  referenceNumber?: string;
+  extraFields?: QuoteExtraFields[];
+}
+
+export interface QuoteInfoAndExtrafieldsItemProps {
+  info: {
+    quoteTitle: string;
+    referenceNumber: string;
+  };
+  extraFields: QuoteExtraFields[] | undefined;
+  recipients: string[];
+}
+
+export interface B2bExtraFieldsProps {
+  defaultValue: string;
+  fieldName: string;
+  fieldType: 0 | 1 | 2 | 3;
+  isRequired: boolean;
+  labelName: string;
+  listOfValue: null | Array<string>;
+  maximumLength: string | number | null;
+  maximumValue: string | number | null;
+  numberOfRows: string | number | null;
+  visibleToEnduser: boolean;
+  id: number | string;
+}
+
+export interface FieldsOptionProps {
+  label: string;
+  value: string | number;
+}
+
+export interface FormattedItemsProps {
+  [key: string]: string | boolean | number | Array<FieldsOptionProps> | boolean | undefined;
+  name: string;
+  default: string | number;
+  id: string | number;
```

Comment:
This is a bad type - too generic

Reply:
Type has been redefined. Thanks for your suggestion